modules/configuration/src/main/java/org/apache/ignite/internal/configuration/ConfigurationChanger.java

```diff
@@ -223,6 +233,11 @@ public ConfigurationChanger(
         this.configurationValidator = configurationValidator;
         this.rootKeys = rootKeys.stream().collect(toMap(RootKey::key, identity()));
         this.migrator = migrator;
+
+        this.deletedPrefixes = deletedPrefixes.stream()
+                .map(deletedKey -> deletedKey.replace(".", "\\.").replace("*", ".*") + ".*")
```

Comment:
Let's assume that I deleted `ignite.foo` and introduced `ignite.fooMillis` few months later. Will the new configuration match the old prefix for deleted configuration? It should not match it. Please add such a test